### PR TITLE
Fix: Allow MaxBreadcrumb 0  / Expose MaxBreadcrumb metadata.

### PR DESCRIPTION
**Pull request by lucas-zimerman**
## :scroll: Description

Previously, when MaxBreadcrumbs was set to 0, the app would crash since the CircularQueue implementation must have a size higher than 0.
With that, I made a `disabledqueue` that behaves similarly like `CircularQueue` do, with the difference that any items inserted into it will be ignored. 

## :bulb: Motivation and Context

I was also surprised to see that max breadcrumbs wasn't exposed on the metadata, so I also exposed it on this PR to make tests with the sample easier.

To fix
https://github.com/getsentry/sentry-react-native/issues/3634
and
https://github.com/getsentry/sentry-java/issues/3313

## :green_heart: How did you test it?

Unit tests and running the sample app with max breadcrumbs set to 0 and 100.

## :pencil: Checklist

- [x] I reviewed the submitted code.
- [x] I added tests to verify the changes.
- [x] No new PII added or SDK only sends newly added PII if `sendDefaultPII` is enabled.
- [ ] I updated the docs if needed.
- [ ] Review from the native team if needed.
- [x] No breaking change or entry added to the changelog.
- [x] No breaking change for hybrid SDKs or communicated to hybrid SDKs.

### CODE REVIEW

File: sentry-android-core/src/test/java/io/sentry/android/core/ManifestMetadataReaderTest.kt
Comment (stefanosiano) [inline sentry-android-core/src/test/java/io/sentry/android/core/ManifestMetadataReaderTest.kt:1533]:
```suggestion
    fun `applyMetadata reads maxBreadcrumbs to options and keeps default if not found`() {
```

**Review by stefanosiano — APPROVED** [review]
looks good!

**Comment by lucas-zimerman** [comment]
@stefanosiano just saying that I cannot merge PRs here, do we need to wait for more approvals before merging it?

File: sentry/src/main/java/io/sentry/DisabledQueue.java
Comment (adinauer) [inline sentry/src/main/java/io/sentry/DisabledQueue.java:37]: @lucas-zimerman @stefanosiano should this return `true` instead? If no, can we document why?

**Pull request merged**
